enterprise/server/secrets/secrets.go

```diff
@@ -150,15 +151,45 @@ func (s *SecretService) UpdateSecret(ctx context.Context, req *skpb.UpdateSecret
 	// the secret box using this group's public key.
 	_, err = keystore.OpenAnonymousSealedBox(s.env, grp.PublicKey, grp.EncryptedPrivateKey, req.GetSecret().GetValue())
 	if err != nil {
-		return nil, err
+		return nil, false, err
 	}
-	err = dbHandle.DB(ctx).Exec(`REPLACE INTO "Secrets" (user_id, group_id, name, value, perms) VALUES (?, ?, ?, ?, ?)`,
-		u.GetUserID(), u.GetGroupID(), req.GetSecret().GetName(), req.GetSecret().GetValue(), secretPerms.Perms).Error
+
+	newSecret := false
+	err = dbHandle.Transaction(ctx, func(tx *gorm.DB) error {
+		rows, err := tx.Raw(`
+			SELECT * 
+			FROM "Secrets" 
+			WHERE group_id = ? AND name = ?
```

Comment:
does this need to be SELECT FOR UPDATE to lock?

Reply:
we add `dbHandle.SelectForUpdateModifier()` below

Reply:
ah got it, forgot that was db specific. thanks!